pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -198,12 +198,112 @@ func (c *Cluster) validateOSDSettings() error {
 	return nil
 }
 
+func (c *Cluster) ValidateTopologyAcrossNodes() error {
+	nodes, err := k8sutil.GetKubernetesNodesMatchingRookNodes(c.clusterInfo.Context, c.spec.Storage.Nodes, c.context.Clientset)
```

Comment:
The `GetKubernetesNodesMatchingRookNodes()` is actually for a specific case where individual nodes are listed under the `storage.nodes` section of the cephcluster CR. Let's just analyze all nodes instead of filtering them. Even if they don't use all the nodes for OSDs, this check can be comprehensive for all nodes. 
```suggestion
	nodes, err := clientset.CoreV1().Nodes().List(c.clusterInfo.Context), metav1.ListOptions{})
```